### DISCUSSION TITLE
Gift button placement in marketplace dialog

### DISCUSSION THREAD

**Discussion by zenseii** [ihhub/fheroes2#4621] [Ideas]
Currently you can only see the gift button in the marketplace dialog if you have either selected two untradeable resources, just arrived or just made a trade. In other words getting to the gift button can be very non-intuitive in certain conditions.

Problematic example: If have enough to exhange every resource, and you open the marketplace and you first want to have a look at how much you maximally can trade for each resource before making a gift, then you will have to either make a simple trade first, select the same resource as to and from, or exit and reopen the marketplace for the gift button to appear. This is very cumbersome.

I have a suggestion that we move the gift button to the bottom like in this mock-up:
![image](https://user-images.githubusercontent.com/12501091/141441904-d9bb2347-df40-4901-80b3-bfde8837543c.png)

Another point aside from this is that it would make sense to have the gift button removed if you're not playing against other humans or if none of the AI are on your team.  In these cases the "Exit" button could be moved back to the center. It does seem like there is already some logic in the code to easily add this.

I'd like to have an opinion from some of the preservationist members of the core team on this, like @LeHerosInconnu .

**Comment by a1exsh** [comment]
> Another point aside from this is that it would make sense to have the gift button removed if you're not playing against other humans or if none of the AI are on your team.

Why remove the possibility to gift some resources to AI it might desperately need? ;)

**Reply by zenseii** [reply]
Or as an added challenge you could send your opponent all your resources in the beginning 😅

**Reply by LeHerosInconnu** [reply]
Hello @zenseii and @a1exsh,

> Currently you can only see the gift button in the marketplace dialog if you have either selected two untradeable resources, just arrived or just made a trade. In other words getting to the gift button can be very non-intuitive in certain conditions.
> 
> Problematic example: If have enough to exhange every resource, and you open the marketplace and you first want to have a look at how much you maximally can trade for each resource before making a gift, then you will have to either make a simple trade first, select the same resource as to and from, or exit and reopen the marketplace for the gift button to appear. This is very cumbersome.
> 
> I have a suggestion that we move the gift button to the bottom like in this mock-up:
> ![image](https://user-images.githubusercontent.com/12501091/141441904-d9bb2347-df40-4901-80b3-bfde8837543c.png)
> 
> Another point aside from this is that it would make sense to have the gift button removed if you're not playing against other humans or if none of the AI are on your team. In these cases the "Exit" button could be moved back to the center. It does seem like there is already some logic in the code to easily add this.
> 
> I'd like to have an opinion from some of the preservationist members of the core team on this, like @LeHerosInconnu .

This would make the GIFT button always accessible, that's a good idea.
I vote for it. :)

> > Another point aside from this is that it would make sense to have the gift button removed if you're not playing against other humans or if none of the AI are on your team.
> 
> Why remove the possibility to gift some resources to AI it might desperately need? ;)

> Or as an added challenge you could send your opponent all your resources in the beginning :sweat_smile:

Personally, I don't use the possibility to give resources.
But it can be a strategy to give resources to the AI to build its castles before you capture them. :D

I think it's okay to let the player have those options.
The player who doesn't need it can just ignore it without changing the way the original game works.

**Reply by zenseii** [reply]
> But it can be a strategy to give resources to the AI to build its castles before you capture them. :D

You have a very sly reasoning and I like it 🤣 (Just don't tell ihhub or he will probably try to adjust the AI for these "strategies"...)

> I think it's okay to let the player have those options. The player who doesn't need it can just ignore it without changing the way the original game works.

I see! Seeing that you are okay with it this way then I'll have a PR up shortly (I've more or less already finished it anyways.)

I also noticed trying to trade between human players in hotseat that gifting doesn't always work. I'll have to research this a bit more, but right now I *think* that any other player needs a marketplace for the gifts to arrive to any player (even one without marketplace.).

**Reply by ihhub** [reply]
AI will abuse this move for sure! I'll make it happen ;)
